config.go

```diff
@@ -65,6 +72,14 @@ func (c *Config) Merge(config *Config) {
 		c.SSLNoVerify = true
 	}
 
+	if (config.HttpAuth.Username != "") || (config.HttpAuth.Password != "") {
+		c.HttpAuth = config.HttpAuth
+	}
+
+	if config.HttpAuth.Password != "" {
```

Comment:
This should probably be all-or-nothing for the best clarity. If user or password are present, then merge, else do nothing.